extensions/arc/src/models/controllerModel.ts

```diff
@@ -91,6 +117,12 @@ export class ControllerModel {
 		}
 	}
 	public async refresh(showErrors: boolean = true, promptReconnect: boolean = false): Promise<void> {
+		//wait for any previous refresh that might be in progress to finish
```

Comment:
While this is ok for now, this synchronization needs to be done at AzdataLogin level so that apis that are not using refresh will also benefit from an assurance that Login context does not change unexpectedly.